lib/jenkins_api_client/job.rb

```diff
@@ -299,6 +299,16 @@ def build_freestyle_config(params)
             xml.description
             xml.keepDependencies "#{params[:keep_dependencies]}"
             xml.properties
+            #buildlogs related stuff
+            if params[:discard_old_builds]
```

Comment:
If I understand correctly, this method `build_freestyle_config` only outputs the XML for a job config, rather than actually creating that job on the server.  If so, this is 👍  to merge, I got response XML like this while using login_with_irb:

```
2.2.7 :001 > @client.job.build_freestyle_config({name: 'testjob', discard_old_builds: {daysToKeep: 3, numToKeep: 1, artifactDaysToKeep: 1, artifactNumToKeep: 1}})
 => "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<project>\n  <actions/>\n  <description/>\n  <keepDependencies>false</keepDependencies>\n  <properties/>\n  <logRotator class=\"hudson.tasks.LogRotator\">\n    <daysToKeep>3</daysToKeep>\n    <numToKeep>1</numToKeep>\n    <artifactDaysToKeep>1</artifactDaysToKeep>\n    <artifactNumToKeep>1</artifactNumToKeep>\n  </logRotator>\n  <scm class=\"hudson.scm.NullSCM\"/>\n  <canRoam>true</canRoam>\n  <disabled>false</disabled>\n  <blockBuildWhenDownstreamBuilding>false</blockBuildWhenDownstreamBuilding>\n  <blockBuildWhenUpstreamBuilding>false</blockBuildWhenUpstreamBuilding>\n  <triggers class=\"vector\"/>\n<concurrentBuild>false</concurrentBuild>\n  <builders/>\n  <publishers/>\n  <buildWrappers/>\n</project>\n"
```

@arangamani as long as I understand this correctly, can you 👍 and merge?

Reply:
@katelovescode Go for it